### ISSUE TITLE
Custom regularizer does not remain the same object

### ISSUE THREAD

**Issue by Mastush** [keras-team/keras#12690]
Short version: Custom regularizer object seems to be reconstructed with each batch.

Full version:
I'm trying to create a custom Keras regularizer that uses the distance of the layer's weights from it's original weights, but what I used doesn't work and I get a zero difference at all times.

This is the regularizer code:

```python
class NormReg():
    def __init__(self, coeff):
        self._coeff = coeff
        self._original_weights = None

    def _norm(self, weight_matrix):
        return K.sum(K.square(weight_matrix))

    def __call__(self, weight_matrix):
        if self._original_weights is None:
            self._original_weights = weight_matrix

        diff_matrix = weight_matrix - self._original_weights
        return self._coeff * self._norm(diff_matrix)
```

(I'm using tensorflow as the backend)

After playing with this class a bit, I noted something strange: It's as if the regularizer object is being created over and over again in the training with each batch, which does explain why I'm getting zeros.
I got to this conclusion by changing the class to -

```python
class NormReg():
    def __init__(self, coeff):
        self._ugly_check = 1
        self._coeff = coeff
        self._original_weights = None

    def _norm(self, weight_matrix):
        return K.sum(K.square(weight_matrix))

    def __call__(self, weight_matrix):
        if self._original_weights is None:
            self._original_weights = weight_matrix
        if self._ugly_check == 1:
            self._ugly_check = 0
            return 10000
        diff_matrix = weight_matrix - self._original_weights
        return self._coeff * self._norm(diff_matrix)
```

And seeing that the loss does in fact suffer the penalty that follows from _ugly_check being 1 throughout the training.

I would expect the regularizer object to remain the same one throughout training. Is this a bug or am I not understanding the usage of custom regularizers correctly?
